action-sonos.py

```diff
@@ -51,19 +52,43 @@
 HERMES_HOST = "{}:1883".format(HOSTNAME)
 
 
+# Hotword callback
+def hotword_detected_callback(hermes, sessionStartedMessage):
+    devices = hermes.device_discovery_service.get_devices()
+
+    hermes.states['devices'] = {device.identifier: device for device in devices}
+
+    for device in hermes.states.get('devices').values():
+        device_at_lowered_volume = copy.deepcopy(device)
+        device_at_lowered_volume.volume = 10
+        hermes.device_transport_control_service.set_volume(device_at_lowered_volume)
+
+
+def restore_volume_for_hotword(intent_callback):
+    def restore_volume_wrapper(hermes, intentMessage):
+        intent_callback(hermes, intentMessage)  # We call the callback
+
+        # We restore the volume to what it was before the hotword was detected.
+        for device in hermes.states.get('devices').values():
+            hermes.device_transport_control_service.set_volume(device)
+
+    return restore_volume_wrapper
+
+
 # Music management functions
+@restore_volume_for_hotword
```

Comment:
Not completely sure of the magic behind this, I am assuming that the `restore_volume_for_hotword` function would be called always before the callback it sits on (in this case `addSong_callback`) and trigger the `restore_volume_wrapper` callback, right?